pkg/engine/k8smanifest.go

```diff
@@ -0,0 +1,401 @@
+	return ruleResp
+}
+
+func handleVerifyManifest(ctx *PolicyContext, rule *kyvernov1.Rule, logger logr.Logger) *response.RuleResponse {
+	verified, reason, err := verifyManifest(ctx, *rule.Validation.Manifests, logger)
+	if err != nil {
+		logger.V(3).Info("verifyManifest return err", "error", err.Error())
+		return ruleError(rule, response.Validation, "error occurred during manifest verification", err)
+	}
+	logger.V(3).Info("verifyManifest result", "verified", strconv.FormatBool(verified), "reason", reason)
+	if !verified {
+		return ruleResponse(*rule, response.Validation, reason, response.RuleStatusFail, nil)
+	}
+	return ruleResponse(*rule, response.Validation, reason, response.RuleStatusPass, nil)
+}
+
+func verifyManifest(policyContext *PolicyContext, verifyRule kyvernov1.Manifests, logger logr.Logger) (bool, string, error) {
+	// load AdmissionRequest
+	request, err := policyContext.JSONContext.Query("request")
+	if err != nil {
+		return false, "", errors.Wrapf(err, "failed to get a request from policyContext")
+	}
+	reqByte, _ := json.Marshal(request)
+	var adreq *admissionv1.AdmissionRequest
+	err = json.Unmarshal(reqByte, &adreq)
+	if err != nil {
+		return false, "", errors.Wrapf(err, "failed to unmarshal a request from requestByte")
+	}
+	// unmarshal admission request object
+	var resource unstructured.Unstructured
+	objectBytes := adreq.Object.Raw
+	err = json.Unmarshal(objectBytes, &resource)
+	if err != nil {
+		return false, "", errors.Wrapf(err, "failed to Unmarshal a requested object")
+	}
+
+	logger.V(4).Info("verifying manifest", "namespace", adreq.Namespace, "kind", adreq.Kind.Kind,
+		"name", adreq.Name, "username", adreq.UserInfo.Username)
+
+	// allow dryrun request
+	if adreq.DryRun != nil && *adreq.DryRun {
+		return true, "allowed because of DryRun request", nil
+	}
+
+	// prepare verifyResource option
+	vo := &k8smanifest.VerifyResourceOption{}
+	// adding default ignoreFields from
+	// github.com/sigstore/k8s-manifest-sigstore/blob/main/pkg/k8smanifest/resources/default-config.yaml
+	vo = k8smanifest.AddDefaultConfig(vo)
+	// adding default ignoreFields from pkg/engine/resources/default-config.yaml
+	vo = addDefaultConfig(vo)
+	// adding ignoreFields from Policy
+	for _, i := range verifyRule.IgnoreFields {
+		converted := k8smanifest.ObjectFieldBinding(i)
+		vo.IgnoreFields = append(vo.IgnoreFields, converted)
+	}
+
+	// dryrun setting
+	vo.DisableDryRun = !verifyRule.DryRunOption.Enable
+	if verifyRule.DryRunOption.Namespace != "" {
+		vo.DryRunNamespace = verifyRule.DryRunOption.Namespace
+	} else {
+		vo.DryRunNamespace = config.KyvernoNamespace()
+	}
+	if !vo.DisableDryRun {
+		// check if kyverno can 'create' dryrun resource
+		ok, err := checkDryRunPermission(policyContext.Client, adreq.Kind.Kind, vo.DryRunNamespace)
+		if err != nil {
+			logger.V(1).Info(fmt.Sprintf("failed to check permissions to 'create' resource %s/%s. disabled DryRun option.: %s", vo.DryRunNamespace, adreq.Kind.Kind, err.Error()))
```

Comment:
Please change log style to remove fmt and instead pass individual arguments (klog / Zap style). This will me helpful when viewing the log as JSON.